azurerm/resource_arm_automation_module_test.go

```diff
@@ -0,0 +1,132 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMAutomationModule_basic(t *testing.T) {
+	resourceName := "azurerm_automation_module.test"
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAutomationModuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAutomationModule_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationModuleExists(resourceName),
```

Comment:
Again, we should be checking the actual state of the resource after a apply not just if it exists.